shared/SecretManager/SecretsStore.cs

```diff
@@ -0,0 +1,90 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Configuration.UserSecrets;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.Extensions.SecretManager
+{
+    /// <summary>
+    /// Provides read and write access to the secrets.json file for local user secrets.
+    /// This is not thread-safe.
+    /// This object is meant to have a short lifetime.
+    /// When calling <see cref="Save"/>, this will overwrite the secrets.json file. It does not check for concurrency issues if another process has edited this file.
+    /// </summary>
+    internal class SecretsStore : IReadOnlyDictionary<string, string>
```

Comment:
ugh - plurals in names always make a mess of things. Can this just be `SecretStore`?